### ISSUE TITLE
Unable to start debugging. An item with the same key has already been added.

### ISSUE THREAD

**Issue by paaaa** [microsoft/vscode-cpptools#2056]
error: Unable to start debugging. An item with the same key has already been added.
* Operating System and version ubuntu 18.04
 VS Code version 1.23.1 commit:  d0182c3417d225529c6d5ad24b7572815d0de9ac
log: 1: (4554) ->(gdb)
1: (4554) 1007: elapsed time 4167
1: (4556) <-1008-interpreter-exec console "show architecture"
1: (4556) ->~"The target architecture is set automatically (currently i386:x86-64)\n"
1: (4556) ->1008^done
1: (4556) ->(gdb)
1: (4556) 1008: elapsed time 0
1: (4558) <-1009-break-insert main
1: (5505) ->1009^done,bkpt={number="1",type="breakpoint",disp="keep",enabled="y",addr="<MULTIPLE>",times="0",original-location="main"},{number="1.1",enabled="y",addr="0x0000000002317ec6",func="main()",file="",fullname="",line="305",thread-groups=["i1"]},{number="1.2",enabled="y",addr="0x000000000234c5ee",func="td::DcId::main()",file="",fullname="",line="20",thread-groups=["i1"]}
1: (5505) ->(gdb)
1: (5507) 1009: elapsed time 948
1: (5513) <--gdb-exit
1: (5513) ->^exit

**Comment by paaaa** [comment]
I can use gdb to debug

**Comment by pieandcakes** [comment]
> error: Unable to start debugging. An item with the same key has already been added.

Can you check your environment variables and see if you have a duplicate with slightly different casing? this is an issue with the Process object we use to spawn processes using .NET or Mono's framework.

**Comment by paaaa** [comment]
echo $PATH
/home/aaaa/tools/jdk-8u171-linux-x64/jdk1.8.0_171/bin:/home/aaaa/tools/android-ndk-r14b-linux-x86_64/android-ndk-r14b:/home/aaaa/Android/Sdk/platform-tools/:/home/aaaa/tools/kotlin-compiler-1.2.41/kotlinc/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

# I don't seem to see other shell

**Comment by paaaa** [comment]
echo $PATH
/home/----/bin:/home/----/tools/jdk-8u171-linux-x64/jdk1.8.0_171/bin:/home/----/tools/android-ndk-r14b-linux-x86_64/android-ndk-r14b:/home/----/Android/Sdk/platform-tools/:/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin

# this my environment variables can you help me

**Comment by paaaa** [comment]
{
    // 使用 IntelliSense 了解相关属性。 
    // 悬停以查看现有属性的描述。
    // 欲了解更多信息，请访问: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "(gdb) Launch",
            "type": "cppdbg",
            "request": "launch",
            "program": "${workspaceFolder}/example/cpp/build/td_example",
            "args": [],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [],
            "externalConsole": true,
            "MIMode": "gdb",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                }
            ],
            "logging": {
                "engineLogging": true
                }
        }
    ]
}

this my launch.json

**Comment by aghpub** [comment]
I'm having the same problem. I'm able to debug a simple "Hello World" program but debugging the attached demo project fails with "Unable to start debugging. An item with the same key has already been added.". Console debugging outside of VScode works.

This happens on Arch Linux using VScode 1.23.1 (Shell 1.7.12, Renderer 58.0.3029.110, Node 7.9.0, Architecture x64) with GDB 8.1. I'll attach the failing project as well as my (redacted) environment to this post. Help would be appreciated.

[problem_2056.zip](https://github.com/Microsoft/vscode-cpptools/files/2075955/problem_2056.zip)

[environment.txt](https://github.com/Microsoft/vscode-cpptools/files/2075881/environment.txt)

EDIT:

Added missing copright notice to 3rd party code included in the attached project

**Comment by aghpub** [comment]
As an update to my comment from yesterday: I upgraded to VScode 1.24.0 and for some reason I was able to debug the example project. I still get the error with another project though. So upgrading VScode did not solve the problem for me.

**Comment by pieandcakes** [comment]
@aghpub, @paaaa can either of you enable `"logging": { "engineLogging": true, "trace": true" }` and send me the output? Trying to see where this error is coming from. I can't seem to duplicate it on my machine.

**Comment by aghpub** [comment]
@pieandcakes I'll attach the log from a failed debugging session. Anyway it might not help since the log looks like everything went ok despite the error popping up

[logging.txt](https://github.com/Microsoft/vscode-cpptools/files/2083490/logging.txt)

**Comment by michaelkourlas** [comment]
@pieandcakes I don't see any evidence of duplicate environment variable names. I've attached some logging output as well from a failed debugging session.

[printenv.txt](https://github.com/Microsoft/vscode-cpptools/files/2084399/printenv.txt)
[logging.txt](https://github.com/Microsoft/vscode-cpptools/files/2085949/logging.txt)

**Comment by pieandcakes** [comment]
@michaelkourlas @aghpub Thank you for the logs. Let me see what is happening.

**Comment by pieandcakes** [comment]
@michaelkourlas Do you have the beginning of the log? I'm missing some key information from what you posted. Can you add `"traceResponse": true` to your logging block? If you don't want it public, can you email it to me?

**Comment by pieandcakes** [comment]
@aghpub I don't see a failure in that log you provided. Did you get the error in that one?

If you have an error case, can you make sure you have `"trace": true, "traceResponse": true` added to your logging blocks and send me that log?

**Comment by aghpub** [comment]
@pieandcakes I added `"traceResponse": true` to launch.json and the log now conains something related to the error. I attach the log to this post.

[logging.txt](https://github.com/Microsoft/vscode-cpptools/files/2085531/logging.txt)

**Comment by michaelkourlas** [comment]
@pieandcakes Sorry, looks like that was cut off by accident. I've updated the comment with the full log.

**Comment by pieandcakes** [comment]
I think I see the error. Trying to determine why it is happening. I'll let you know as soon as I know what is causing it.

**Comment by noodlecollie** [comment]
I'm also getting this error.

VS Code 1.24.0 x64 6a6e02cef0f2122ee1469765b704faf5d0e0d859
Manjaro (Arch) Linux 17.1.10

Just started happening today on a new project I'm working on, but works fine on other projects. Log is [attached](https://github.com/Microsoft/vscode-cpptools/files/2088006/log.txt), launch.json is as follows:

```
{
	"version": "0.2.0",
	"configurations": [
		{
			"name": "(gdb) Launch",
			"type": "cppdbg",
			"request": "launch",
			"program": "${workspaceFolder}/build/bin/calliper",
			"args": [],
			"stopAtEntry": false,
			"cwd": "${workspaceFolder}/build/bin",
			"environment": [],
			"externalConsole": true,
			"MIMode": "gdb",
			"setupCommands": [
				{
					"description": "Enable pretty-printing for gdb",
					"text": "-enable-pretty-printing",
					"ignoreFailures": true
				}
			],
			"logging":{
				"engineLogging": true,
				"trace": true,
				"traceResponse": true
			}
		}
	]
}
```

**Comment by pieandcakes** [comment]
@x6herbius thanks for your log. 
The problem is related to code setup to track our entrypoint breakpoint. We issue a command of `-break-insert main` and the assumption was made this would return one bound location but in all these logs its showing us multiple breakpoint bind locations. I'm working on a fix right now. Unfortunately there is no workaround for this issue.

**Comment by noodlecollie** [comment]
Thanks for the timely response! I'll keep an eye out - luckily for the moment I can manage without the VSCode debugger.

**Comment by pieandcakes** [comment]
I have a possible fix. When it is approved, I'll put together a build for you to verify that it unblocks this issue. 

Thanks for your patience.

**Comment by pieandcakes** [comment]
Please try this release of the extension and let me know if it works. Rename the file cpptools.vsix, uninstall the current version and reload VS Code. Then use the following menu on the Extensions tab to install the file:

![image](https://user-images.githubusercontent.com/17558378/41321089-39d868b2-6e57-11e8-8ea1-bd3678a8b2f2.png)

Fixed extension:
[cpptools.zip](https://github.com/Microsoft/vscode-cpptools/files/2096026/cpptools.zip)

**Comment by aghpub** [comment]
The new version seems to fix the problem for me. :-)

**Comment by noodlecollie** [comment]
Will give this a try this evening when I'm home!

**Comment by Wyverex42** [comment]
I had the same problem and the fix works for me. Thank you!

**Comment by pieandcakes** [comment]
Please use this until we ship our next update.

**Comment by noodlecollie** [comment]
Unfortunately this doesn't fix the issue for me. Log file is attached: [log-13-06-18.txt](https://github.com/Microsoft/vscode-cpptools/files/2099483/log-13-06-18.txt)

**Comment by pieandcakes** [comment]
@x6herbius Can you make sure that you installed the package linked here? I used your log and I can't duplicate that error.

Make sure you are choosing `uninstall` then `reload` and installing this package after VS Code has reloaded.

**Comment by noodlecollie** [comment]
Hmm, not quite sure what happened the first time - I even uninstalled and reinstalled to make sure - but after doing it a third time it does seem to work. Maybe my VSCode is just playing with me. Thanks for the patch!

**Comment by cksmith** [comment]
I believe I installed the updated extension properly but I seem to still be having the same problem as the others. I uninstalled, reloaded, then installed the via the .vsix file. Can you please take a look at the log and see if the problem is the same?
[log-06-21-2018.txt](https://github.com/Microsoft/vscode-cpptools/files/2125197/log-06-21-2018.txt)

**Comment by pieandcakes** [comment]
@cksmith yes, you have the same issue. Please uninstall the extension, close VS Code, go to your ~/.vscode/extensions folder and delete the cpptools folder and then install the fixed vsix through the `...` option under Extensions in VS Code.

**Comment by cksmith** [comment]
@pieandcakes Deleting that folder and reinstalling did the trick, thanks!.

**Comment by sean-mcmanus** [comment]
Should be fixed with 0.17.5.
